### ISSUE TITLE
Error on install after 3.1 update

### ISSUE THREAD

**Issue by janhenckens** [fruitstudios/craft-linkit#50]
When installing  & activating the plugin on a 3.1 site, we get the following error on the plugin overview screen:
```
Invalid plugin – craft\errors\InvalidPluginException
No plugin exists with the handle "fruit-link-it".
````

This happens during an update from 2.7.4 to 3.1.3.
Tried running the entire migration/update again with the plugin installed but same error.

**Comment by pgrzyb** [comment]
Same issue here - any news on this? 
I can provide a stack trace if that helps:
`2019-01-28 21:58:02 [-][5][lfl50ofbpqeijr6mmvk79nn0gi][error][craft\errors\InvalidPluginException] craft\errors\InvalidPluginException: No plugin exists with the handle "fruit-link-it". in /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/services/Plugins.php:1337
Stack trace:
#0 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/services/Plugins.php(1295): craft\services\Plugins->_getPluginConfigData('fruit-link-it')
#1 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/services/Plugins.php(935): craft\services\Plugins->_loadDisabledPluginInfo()
#2 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/services/Plugins.php(912): craft\services\Plugins->getPluginInfo('super-table')
#3 /home/vagrant/Code/site/craft3/vendor/twig/twig/lib/Twig/Extension/Core.php(1626): craft\services\Plugins->getAllPluginInfo()
#4 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/helpers/Template.php(73): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig_Source), Object(craft\services\Plugins), 'getAllPluginInf...', Array, 'method', false, false)
#5 /home/vagrant/Code/site/craft3/storage/runtime/compiled_templates/0b/0b00554165ec5e721954bb32928f394984c2c9e9ed5b1a248c62194633b0709e.php(37): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig_Source), Object(craft\services\Plugins), 'getAllPluginInf...', Array, 'method')
#6 /home/vagrant/Code/site/craft3/vendor/twig/twig/lib/Twig/Template.php(386): __TwigTemplate_4147eceeafd02d451e89c2b8def4ae4f5cea1678fd43608d5f9d27a6b8a24e1d->doDisplay(Array, Array)
#7 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/web/twig/Template.php(49): Twig_Template->displayWithErrorHandling(Array, Array)
#8 /home/vagrant/Code/site/craft3/vendor/twig/twig/lib/Twig/Template.php(363): craft\web\twig\Template->displayWithErrorHandling(Array, Array)
#9 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/web/twig/Template.php(31): Twig_Template->display(Array, Array)
#10 /home/vagrant/Code/site/craft3/vendor/twig/twig/lib/Twig/Template.php(371): craft\web\twig\Template->display(Array)
#11 /home/vagrant/Code/site/craft3/vendor/twig/twig/lib/Twig/Environment.php(289): Twig_Template->render(Array)
#12 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/web/View.php(337): Twig_Environment->render('settings/plugin...', Array)
#13 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/web/View.php(384): craft\web\View->renderTemplate('settings/plugin...', Array)
#14 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/web/Controller.php(161): craft\web\View->renderPageTemplate('settings/plugin...', Array)
#15 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/controllers/TemplatesController.php(78): craft\web\Controller->renderTemplate('settings/plugin...', Array)
#16 [internal function]: craft\controllers\TemplatesController->actionRender('settings/plugin...', Array)
#17 /home/vagrant/Code/site/craft3/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#18 /home/vagrant/Code/site/craft3/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#19 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/web/Controller.php(109): yii\base\Controller->runAction('render', Array)
#20 /home/vagrant/Code/site/craft3/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('render', Array)
#21 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/web/Application.php(297): yii\base\Module->runAction('templates/rende...', Array)
#22 /home/vagrant/Code/site/craft3/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('templates/rende...', Array)
#23 /home/vagrant/Code/site/craft3/vendor/craftcms/cms/src/web/Application.php(286): yii\web\Application->handleRequest(Object(craft\web\Request))
#24 /home/vagrant/Code/site/craft3/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#25 /home/vagrant/Code/site/craft3/web/index.php(21): yii\base\Application->run()
#26 {main}`

**Comment by bfopma** [comment]
I'm experiencing the same issue.

**Comment by bossanova808** [comment]
And me!

**Comment by pgrzyb** [comment]
TL;DR Is there any chance we could get a status on this? 

Or at least an update on whether it will or will not be attacked in the near future? I am semi stuck moving a project to Craft 3, and as much as it would be a huge annoyance for me to redo all the content not to use linkit, should you not have the time or resources to fix this, that's what I'll have to do :-(

**Comment by samhibberd** [comment]
Thanks all for your patience, we are aware of the issue are hoping to have a fix released imminently.

**Comment by bfopma** [comment]
Negatory—no Super Table install here.

**Comment by darylknight** [comment]
Happening for me too - went from Craft 2.7.4 (LinkIt 2.3.4) to Craft 3.1.7 (LinkIt 1.1.6).
I don't have SuperTable installed on the old or new version of the site.

**Comment by samhibberd** [comment]
Thanks. I'm currently rewriting the migrations based some new project config stuff from the folks at Craft. Hoping to get an update release out today.

**Comment by pgrzyb** [comment]
That's some amazing news! Thanks for looking into this @samhibberd - looking forward to the update 👍

**Comment by bossanova808** [comment]
I do use supertable.

Sent on the go...

________________________________

**Comment by darylknight** [comment]
Any luck @samhibberd ? I hate to rush you but my deadline for upgrading Craft 3 is this week, and I currently can't access the front end of the site because the field doesn't exist.

**Comment by bfopma** [comment]
> Any luck @samhibberd ? I hate to rush you but my deadline for upgrading Craft 3 is this week, and I currently can't access the front end of the site without commenting out all the linkit fields, and I can't access the plugin store or the plugins section of the control panel.

@darylknight I'm not sure if this is an option for you, but in my journey to get Link It migrated from Craft 2 -> 3.1 I had to update to Craft 3.0.40 first (using Link It 1.0.7) and then upgrade from there to Craft 3.1 (with Link It ^1.0.7). Tested this today and it worked.

**Comment by samhibberd** [comment]
@bfopma 's workaround should do it, we are are working through some issues with the migration from Craft 2 straight to 3.1+. Really hoping to have a version out today which addresses this.

**Comment by pgrzyb** [comment]
I tried a similar approach, i updated Craft to 3.0.37 first, but sadly Linkit 1.0.6 and 1.0.7 was throwing errors during installation on Craft 3.0.37, so I had to go with Linkit 1.0.5. That worked just fine, no errors, no problems, just joy, and overall happiness.

**Comment by darylknight** [comment]
No luck I'm afraid - I'm on Craft 3.1.8, and every version of LinkIt in the releases section is giving the same error. in `admin/settings/plugins`.

If I try to downgrade Craft to 3.0.40, I get

    Unknown Property – yii\base\UnknownPropertyException
    Setting unknown property: craft\models\Info::config

**Comment by darylknight** [comment]
Adding some more info - I started my upgrade process again this morning from Craft 2.7.4 to 3.0.40. I tried using LinkIt 1.0.5, and although I'm not getting any control panel errors, my fields aren't recognising LinkIt:

![image](https://user-images.githubusercontent.com/1250017/52330704-bf559600-29fe-11e9-841d-47940eb0fd7a.png)

![image](https://user-images.githubusercontent.com/1250017/52330720-c67ca400-29fe-11e9-951f-624c4ce0d96d.png)

**Comment by samhibberd** [comment]
1.1.9 should address this issue, please let us know if you still run into issues.
